host.go

```diff
@@ -58,7 +58,30 @@ type Host struct {
 
 // NewHost creates a Host on top of an existing listener.
 func NewHost(listener *sshd.SSHListener, auth *Auth) *Host {
-	room := chat.NewRoom()
+
+	checkName := func(user *message.User) error {
```

Comment:
Would be nice to pull this out into a helper rather than inline it, just to keep the `NewHost` function easier to follow.

Reply:
I thought the same; I've got an implementation of this in progress (unpushed). Look for it soon.